core/arch/arm/plat-marvell/armada7080/hal_sec_perf.c

```diff
@@ -0,0 +1,304 @@
+		}	\
+	} while (0)
+
+#define RANGE_CODE_TO_SIZE_K(code, sizek)	\
+	((sizek) = ((4) << (code)))	\
+
+#define TZ_LOCK_MC(x)		\
+	do {	\
+		(x) = read32(MCU_MC_CONTROL_0_REG);	\
+		(x) |= (TRUSTZONE_LOCK);	\
+		 write32((x), MCU_MC_CONTROL_0_REG);	\
+	} while (0)
+
+#define _IS_ALIGNED(_addr, _algn)	(!((_addr) & ((_algn) - 1)))
+
+register_phys_mem(MEM_AREA_IO_SEC, MCU_BASE, CORE_MMU_DEVICE_SIZE);
+register_phys_mem(MEM_AREA_IO_SEC, MC_SCR_REGISTER, CORE_MMU_DEVICE_SIZE);
+
+static int32_t _find_valid_range(void)
+{
+	uint32_t i;
+	uint32_t tmp;
+	uint32_t valid;
+
+	for (i = 0; i < MAX_RANGE_NUM; i++) {
+		tmp = read32(MCU_TZ_RANGE_LOW_REG(i));
+		TZ_GET_VALID(tmp, valid);
+		if (!(valid))
+			return i;
+	}
+	return -1;
+}
+
+static int32_t set_range(uint32_t addr, uint32_t size, uint32_t perm)
+{
+	uint32_t data;
+	uint32_t sizecode;
+	int32_t valid_range;
+	uint32_t i;
+
+	if (!_IS_ALIGNED(addr, SIZE_1M)) {
+		EMSG("ERR: region addr(0x%x) is not aligned with 1M!", addr);
+		return -1;
+	}
+
+	if (!_IS_ALIGNED(size, SIZE_1M)) {
+		EMSG("ERR: region size(0x%x) is not aligned with 1M!", size);
+		return -1;
+	}
+
+	if (!_IS_ALIGNED(addr, size)) {
+		EMSG("ERR: region size(0x%x) not align with addr(0x%x)"
+			, size, addr);
+		return -1;
+	}
+
+	RANGE_SIZE_TO_CODE(size, sizecode, i);
+	if (sizecode == INVALID_SIZE_CODE) {
+		EMSG("ERR: not valid region size(2^n)! size:0x%x", size);
+		return -1;
+	}
+
+	valid_range = _find_valid_range();
+	if (valid_range == -1) {
+		EMSG("ERR: can't find valid range!");
+		return -1;
+	}
+
+	data = read32(MCU_TZ_RANGE_LOW_REG(valid_range));
+
+	TZ_SET_VALID(data, 1);
+	TZ_SET_PERM(data, perm);
+	TZ_SET_AREA_LEN_CODE(data, sizecode);
+	TZ_SET_START_ADDR_L(data, addr);
+
+	if (valid_range == 0) {
+		/* Set Undefine Range RW */
+		TZ_SET_UR_PERM(data, RW_PERM);
+		TZ_SET_UR_RZ_EN(data, 0);
+	}
+
+	write32(data, MCU_TZ_RANGE_LOW_REG(valid_range));
+
+	return 0;
+}
+
+static void  _dump_range(void)
+{
+	uint32_t i;
+	uint32_t tmp;
+	uint32_t sizek, __maybe_unused sizem;
+	uint32_t __maybe_unused addr_read, __maybe_unused perm_read;
+	uint32_t valid_read, sizecode_read;
+
+	for (i = 0; i < MAX_RANGE_NUM; i++) {
+		tmp = read32(MCU_TZ_RANGE_LOW_REG(i));
+
+		TZ_GET_VALID(tmp, valid_read);
+		TZ_GET_PERM(tmp, perm_read);
+		TZ_GET_AREA_LEN_CODE(tmp, sizecode_read);
+		TZ_GET_START_ADDR_L(tmp, addr_read);
+
+		if (valid_read) {
+			DMSG("Range Num%d: Reg 0x%lx = 0x%x"
+				, i, MCU_TZ_RANGE_LOW_REG(i), tmp);
+			DMSG("AddrL: 0x%08x", addr_read);
+			RANGE_CODE_TO_SIZE_K(sizecode_read, sizek);
+			sizem = (sizek >> 10);
+			DMSG("Size: %dK, %dM", sizek, sizem);
+			DMSG("Perm: %d", perm_read);
+		}
+	}
+}
+
+static uint32_t _find_granule(uint32_t addr, uint32_t size)
+{
+	/* max supported granule for armada is 8TB
+	   * but 2GB is far enough here
+	   */
+	uint32_t max_granule = 0x10000000;
+
+	while (max_granule >= 1000) {	/* min granule is 4kB */
+		if (max_granule <= size
+			&& _IS_ALIGNED(addr, max_granule)) {
```

Comment:
minor: prefer indentation that distinguish the condition(s) (using spaces) from the block sequence (tabulations).
I.e:
```c
		if (max_granule <= size
		    && _IS_ALIGNED(addr, max_granule))
			return max_granule;
```
Note: the condition fits in a single line
```
		if (max_granule <= size && _IS_ALIGNED(addr, max_granule))
			return max_granule;
```